lib/Solver/CexCachingSolver.cpp

```diff
@@ -132,7 +132,8 @@ struct isValidOrSatisfyingResponse {
   isValidOrSatisfyingResponse(KeyType &_key) : key(_key) {}
 
   bool operator()(ref<SolverResponse> a) const {
```

Comment:
Split `keys` in boolean keys and non-boolean keys and update the predicate accordingly